builtin/providers/aws/resource_aws_elasticache_parameter_group.go

```diff
@@ -0,0 +1,214 @@
+		return fmt.Errorf("Unable to find Parameter Group: %#v", describeResp.CacheParameterGroups)
+	}
+
+	d.Set("name", describeResp.CacheParameterGroups[0].CacheParameterGroupName)
+	d.Set("family", describeResp.CacheParameterGroups[0].CacheParameterGroupFamily)
+	d.Set("description", describeResp.CacheParameterGroups[0].Description)
+
+	// Only include user customized parameters as there's hundreds of system/default ones
+	describeParametersOpts := elasticache.DescribeCacheParametersInput{
+		CacheParameterGroupName: aws.String(d.Id()),
+		Source:               aws.String("user"),
+	}
+
+	describeParametersResp, err := conn.DescribeCacheParameters(&describeParametersOpts)
+	if err != nil {
+		return err
+	}
+
+	d.Set("parameter", flattenElastiCacheParameters(describeParametersResp.Parameters))
+
+	return nil
+}
+
+func resourceAwsElasticacheParameterGroupUpdate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).elasticacheconn
+
+	d.Partial(true)
+
+	if d.HasChange("parameter") {
+		o, n := d.GetChange("parameter")
+		if o == nil {
+			o = new(schema.Set)
+		}
+		if n == nil {
+			n = new(schema.Set)
+		}
+
+		os := o.(*schema.Set)
+		ns := n.(*schema.Set)
+
+		// Expand the "parameter" set to aws-sdk-go compat []elasticacheconn.Parameter
+		parameters, err := expandElastiCacheParameters(ns.Difference(os).List())
+		if err != nil {
+			return err
+		}
+
+		if len(parameters) > 0 {
+			modifyOpts := elasticache.ModifyCacheParameterGroupInput{
+				CacheParameterGroupName: aws.String(d.Get("name").(string)),
+				ParameterNameValues:     parameters,
+			}
+
+			log.Printf("[DEBUG] Modify Cache Parameter Group: %#v", modifyOpts)
+			_, err = conn.ModifyCacheParameterGroup(&modifyOpts)
+			if err != nil {
+				return fmt.Errorf("Error modifying Cache Parameter Group: %s", err)
+			}
+		}
+		d.SetPartial("parameter")
+	}
+
+	d.Partial(false)
+
+	return resourceAwsElasticacheParameterGroupRead(d, meta)
+}
+
+func resourceAwsElasticacheParameterGroupDelete(d *schema.ResourceData, meta interface{}) error {
+	stateConf := &resource.StateChangeConf{
+		Pending:    []string{"pending"},
+		Target:     "destroyed",
+		Refresh:    resourceAwsElasticacheParameterGroupDeleteRefreshFunc(d, meta),
+		Timeout:    3 * time.Minute,
+		MinTimeout: 1 * time.Second,
+	}
+	_, err := stateConf.WaitForState()
+	return err
+}
+
+func resourceAwsElasticacheParameterGroupDeleteRefreshFunc(
+	d *schema.ResourceData,
+	meta interface{}) resource.StateRefreshFunc {
+	conn := meta.(*AWSClient).elasticacheconn
+
+	return func() (interface{}, string, error) {
+
+		deleteOpts := elasticache.DeleteCacheParameterGroupInput{
+			CacheParameterGroupName: aws.String(d.Id()),
+		}
+
+		if _, err := conn.DeleteCacheParameterGroup(&deleteOpts); err != nil {
+			elasticahceerr, ok := err.(awserr.Error)
+			if !ok {
+				return d, "error", err
+			}
+
+			if elasticahceerr.Code() != "CacheParameterGroupNotFoundFault" {
+				return d, "error", err
+			}
+		}
```

Comment:
if `ok` here, but not `Code() != "CacheParameterGroupNotFoundFault"`, we fall through to `return d, "destroyed", nil` , yeah? 

We should probably re-write this that `if ok` do the `Code()` check, otherwise return `return d, "error", err`